### PR TITLE
Block usage of 'Self' and 'Parent' within named formulas

### PR DESCRIPTION
**Pull request by VamsiModem**
Added errors on Self and Parent nodes, If there are any within named formulas.

### CODE REVIEW

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Comment by MikeStall** [comment]
>                 if (!_nameResolver.LookupSelf(out var lookupInfo))

why isn't this check sufficient?


---
Refers to: src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs:3163 in ca0cbe7. [](commit_id = ca0cbe7063a9d143a754679bacbdf8b14e9b5661, deletion_comment = False)

File: src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs [hunk 1]
```diff
@@ -3130,8 +3130,11 @@ public override void Visit(ParentNode node)
                     return;
                 }
 
-                if (!(_nameResolver.CurrentEntity is IExternalControl) || !_nameResolver.LookupParent(out var lookupInfo))
+                if (!(_nameResolver.CurrentEntity is IExternalControl) || 
+                    !_nameResolver.LookupParent(out var lookupInfo) ||
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs:3134]: >_nameResolver.LookupParent(out var lookupInfo)  [](http://example.com/codeflow?start=21&length=47)

Why isn't this check sufficient?

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Comment by VamsiModem** [comment]
We can do this in PAClient without making any changes to PowerFx.

**Pull request closed without merging**